src/ln/channel.rs

```diff
@@ -915,30 +915,23 @@ impl Channel {
 		transaction_utils::sort_outputs(&mut txouts);
 
 		let mut outputs: Vec<TxOut> = Vec::with_capacity(txouts.len());
-		let mut htlcs_included: Vec<HTLCOutputInCommitment> = Vec::with_capacity(txouts.len());
-		let mut htlc_sources: Vec<(PaymentHash, &HTLCSource, Option<u32>)> = Vec::with_capacity(txouts.len() + unincluded_htlc_sources.len());
-		for (idx, out) in txouts.drain(..).enumerate() {
+		let mut htlcs_included: Vec<(HTLCOutputInCommitment, Option<&HTLCSource>)> = Vec::with_capacity(txouts.len() + unincluded_htlc_sources.len());
+		for (idx, mut out) in txouts.drain(..).enumerate() {
 			outputs.push(out.0);
-			if let Some((mut htlc, source_option)) = out.1 {
+			if let Some((mut htlc, source_option)) = out.1.take() {
 				htlc.transaction_output_index = Some(idx as u32);
-				if let Some(source) = source_option {
-					htlc_sources.push((htlc.payment_hash, source, Some(idx as u32)));
-				}
-				htlcs_included.push(htlc);
-			}
-		}
-		for (htlc, source_option) in unincluded_htlc_sources.drain(..) {
-			if let Some(source) = source_option {
-				htlc_sources.push((htlc.payment_hash, source, None));
+				htlcs_included.push((htlc, source_option));
 			}
 		}
+		let non_dust_htlc_count = htlcs_included.len();
+		htlcs_included.append(&mut unincluded_htlc_sources);
```

Comment:
Yea, that was unclear. I guess my change to call "unincluded_htlc_sources"->"included_dust_htlcs" makes it more clear? I consider dust HTLCs "included" since they effect the to_self/remote outputs and are distinct from HTLCs which were not included due to being in a state where we dont care about them.

Reply:
Ah okay if you consider them included because of them being part of miner fee maybe add a hint about it in function comment.

Reply:
Added a comment describing return value to build_commitment_transaction, which was completely absent previously.